File: frontend/scripts/generate-favicons.py
#!/usr/bin/env python3
"""
Generate favicon files from SVG source
Requires: pip install cairosvg pillow
"""

import os
from pathlib import Path
try:
    import cairosvg
    from PIL import Image
except ImportError:
    print("Please install required packages: pip install cairosvg pillow")
    exit(1)

# Define paths
SCRIPT_DIR = Path(__file__).parent
PUBLIC_DIR = SCRIPT_DIR.parent / "public"
SVG_PATH = PUBLIC_DIR / "favicon.svg"

# Define sizes to generate
SIZES = [
    (16, "favicon-16x16.png"),
    (32, "favicon-32x32.png"),
    (180, "apple-touch-icon.png"),
    (192, "android-chrome-192x192.png"),
    (512, "android-chrome-512x512.png"),
]

def generate_favicons():
    """Generate PNG favicons from SVG source"""
    
    if not SVG_PATH.exists():
        print(f"Error: SVG file not found at {SVG_PATH}")
        return
    
    print(f"Generating favicons from {SVG_PATH}")
    
    for size, filename in SIZES:
        output_path = PUBLIC_DIR / filename
        
        # Convert SVG to PNG
        cairosvg.svg2png(
            url=str(SVG_PATH),
            write_to=str(output_path),
            output_width=size,
            output_height=size
        )
        
        print(f"✓ Generated {filename} ({size}x{size})")
    
    # Also create a favicon.ico with multiple sizes
    ico_path = PUBLIC_DIR / "favicon.ico"
    
    # Load different sizes for ICO file
    images = []
    for size in [16, 32, 48]:
        png_path = PUBLIC_DIR / f"favicon-{size}x{size}.png"
        if not png_path.exists():
            # Generate if not exists
            cairosvg.svg2png(
                url=str(SVG_PATH),
                write_to=str(png_path),
                output_width=size,
                output_height=size
            )
        
        img = Image.open(png_path)
        images.append(img)
    
    # Save as ICO
    images[0].save(
        ico_path,
        format="ICO",
        sizes=[(16, 16), (32, 32), (48, 48)]
    )
    print(f"✓ Generated favicon.ico")
    
    print("\nAll favicons generated successfully!")

if __name__ == "__main__":
    generate_favicons()

File: frontend/scripts/generate-favicons.js
const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

// This script requires: npm install sharp

const publicDir = path.join(__dirname, '..', 'public');
const svgPath = path.join(publicDir, 'favicon.svg');

// Read the SVG file
const svgBuffer = fs.readFileSync(svgPath);

// Define sizes to generate
const sizes = [
  { size: 16, name: 'favicon-16x16.png' },
  { size: 32, name: 'favicon-32x32.png' },
  { size: 180, name: 'apple-touch-icon.png' },
  { size: 192, name: 'android-chrome-192x192.png' },
  { size: 512, name: 'android-chrome-512x512.png' }
];

async function generateFavicons() {
  console.log('Generating favicons from SVG...');
  
  for (const { size, name } of sizes) {
    const outputPath = path.join(publicDir, name);
    
    await sharp(svgBuffer)
      .resize(size, size)
      .png()
      .toFile(outputPath);
    
    console.log(`✓ Generated ${name} (${size}x${size})`);
  }
  
  console.log('\nAll favicons generated successfully!');
}

generateFavicons().catch(console.error);